src/software/jetson_nano/services/network/network.cpp

```diff
@@ -22,9 +22,18 @@ std::tuple<TbotsProto::PrimitiveSet, TbotsProto::World> NetworkService::poll(
     const TbotsProto::RobotStatus& robot_status)
 {
     std::scoped_lock lock{primitive_set_mutex, world_mutex};
-    TbotsProto::RobotStatus new_status = robot_status;
-    new_status.set_last_handled_primitive_set(primitive_set_msg.sequence_number());
-    sender->sendProto(robot_status);
+    // Rate limit sending of proto based on thunderloop freq
+    if ((robot_status.motor_status().front_left().motor_faults_size() > 0 ||
+         robot_status.motor_status().front_right().motor_faults_size() > 0 ||
+         robot_status.motor_status().back_left().motor_faults_size() > 0 ||
+         robot_status.motor_status().back_right().motor_faults_size() > 0) ||
+        (network_ticks / (thunderloop_ticks + 1.0) <=
+         ROBOT_STATUS_TO_THUNDERLOOP_HZ_RATIO))
```

Comment:
We should also check whether breakbeam has changed state (Tripped -> Not Tripped or viseversa) I think.

Reply:
added